app/ai_providers/gemini_service.py

```
@@ -69,106 +119,29 @@ def get_gemini_insights(
     try:
-        # ✅ Generate content
         response = gemini_model.generate_content(prompt_template)
         raw_response = response.text.strip()
-        logger.info(f"Gemini raw response for user {user_id}: {raw_response[:300]}...")
 
-        # ✅ STEP 1: Extract ONLY the JSON block between first { and last }
         start_json = raw_response.find('{')
         end_json = raw_response.rfind('}') + 1
-
         if start_json == -1 or end_json == -1:
             raise ValueError("No JSON object found in Gemini response.")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)